springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,420 @@
+public class GtfsValidatorController {
+
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  static final String VALIDATION_REPORT_BUCKET_NAME_ENV_VAR = "VALIDATION_REPORT_BUCKET";
+  private static final String DEFAULT_OUTPUT_BASE = "output";
+  private static final String DEFAULT_NUM_THREADS = "8";
+  private static final String DEFAULT_BUCKET_LOCATION = "US";
+  private static final String MESSAGE_JSON_KEY = "message";
+  private static final String STATUS_JSON_KEY = "status";
+  private static final String DEFAULT_COUNTRY_CODE = "ZZ";
+  private static final String TYPE = "TYPE";
+  private static final String PROJECT_ID = "PROJECT_ID";
+  private static final String PRIVATE_KEY_ID = "PRIVATE_KEY_ID";
+  private static final String PRIVATE_KEY = "PRIVATE_KEY";
+  private static final String CLIENT_EMAIL = "CLIENT_EMAIL";
+  private static final String CLIENT_ID = "CLIENT_ID";
+  private static final String AUTH_URI = "AUTH_URI";
+  private static final String TOKEN_URI = "TOKEN_URI";
+  private static final String AUTH_PROVIDER_X509_CERT_URL = "AUTH_PROVIDER_X509_CERT_URL";
+  private static final String CLIENT_X509_CERT_URL = "CLIENT_X509_CERT_URL";
+
+  /**
+   * Runs {@code org.mobilitydata.gtfsvalidator.cli.Main.main} with the arguments extracted via the
+   * query parameters. Returns the the {@code ResponseEntity} that contains information about the
+   * response's {@code HttpStatus} and a message (as a {@code String}) that gives more information
+   * about success or failure. Please note that this method requires authentication to be setup
+   * prior to execution i.e. the following environment variables have to be defined:
+   *
+   * <ul>
+   *   <li>TYPE
+   *   <li>PROJECT_ID
+   *   <li>PRIVATE_KEY_ID
+   *   <li>PRIVATE_KEY
+   *   <li>CLIENT_EMAIL
+   *   <li>CLIENT_ID
+   *   <li>AUTH_URI
+   *   <li>TOKEN_URI
+   *   <li>AUTH_PROVIDER_X509_CERT_URL
+   *   <li>CLIENT_X509_CERT_URL
+   *   <li>VALIDATION_REPORT_BUCKET
+   * </ul>
+   *
+   * <p>Please check:
+   *
+   * <ul>
+   *   <li><a href=https://cloud.google.com/storage/docs/naming-buckets>this documentation for
+   *       bucket naming guidelines</a>
+   *   <li><a
+   *       href=https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable>this
+   *       documentation for authentication procedure</a>
+   * </ul>
+   *
+   * <a href=https://cloud.google.com/storage/docs/json_api/v1/status-codes>Possible HTTP status
+   * codes</a>
+   *
+   * @param output_base Base directory to store the outputs
+   * @param threads Number of threads to use
+   * @param country_code Country code of the feed, e.g., `nl`. It must be a two-letter country code
+   *     (ISO 3166-1 alpha-2)")
+   * @param url The fully qualified URL to download GTFS archive
+   * @param validation_report_name The name of the validation report including .json extension.
+   * @param dataset_id the id of the dataset validated
```

Comment:
What is this `dataset_id` and where do I find it?

Reply:
Additional documentation has been provided in https://github.com/MobilityData/gtfs-validator/pull/933/commits/3bfbdda1e34f02b5979c7884920ae6ba3ad85aa5.